app/controllers/deploys_controller.rb

```diff
@@ -98,6 +98,7 @@ def changeset
   def destroy
     if @deploy.can_be_stopped_by?(current_user)
       @deploy.stop!
+      SseRailsEngine.send_event('deploys', { type: 'finish' })
```

Comment:
seems strange that these things are spread out into multiple files, maybe handle everything in the deploy ?